src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -0,0 +1,81 @@
+package cromwell.engine
+
+import cromwell.binding.types.WdlType
+import cromwell.binding._
+
+import scala.collection.mutable
+
+
+case class SymbolStoreKey(scope: String, name: String, iteration: Option[Int], input: Boolean)
+
+object SymbolStoreEntry {
+  def buildCallInputEntry(call: Call, inputName: String, wdlType: WdlType): SymbolStoreEntry = {
+    val key = SymbolStoreKey(call.fullyQualifiedName, inputName, iteration = None, input = true)
+    new SymbolStoreEntry(key, wdlType, None)
+  }
+
+  def buildCallOutputEntry(call: Call, taskOutput: TaskOutput): SymbolStoreEntry = {
+    val key = SymbolStoreKey(call.fullyQualifiedName, taskOutput.name, iteration = None, input = false)
+    new SymbolStoreEntry(key, taskOutput.wdlType, None)
+  }
+}
+
+case class SymbolStoreEntry(key: SymbolStoreKey, wdlType: WdlType, wdlValue: Option[WdlValue]) {
+
+  wdlValue match {
+    case Some (value) => wdlType.checkCompatible(value.value)
+    case None => // Nothing is always okay for a value.
+  }
+
+  def isInput: Boolean = key.input
+
+  def isOutput: Boolean = !isInput
+
+  /**
+   * Create a copy of the receiver with `wdlValue` set to `newValue`.
+   */
+  def copyWithUpdatedValue(newValue: WdlValue): SymbolStoreEntry =
+    new SymbolStoreEntry(key, wdlType, Option(newValue))
+
+  def scope: String = key.scope
+}
+
+class SymbolStore(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]) {
+  private val table = mutable.Set[SymbolStoreEntry]()
```

Comment:
Probably should rename `table` too... I don't have any good suggestions... `store`, `symbolSet`?